application/controllers/admin/tokens.php

```diff
@@ -2769,4 +2797,58 @@ protected function tokenIsSetInEmailCache($iSurveyId, $tid, $bIsInvitation)
             && isset($_SESSION[$cacheName][$sType][$tid])
             && $_SESSION[$cacheName][$sType][$tid] > 0;
     }
+
+    /**
+     * Dispatches the beforeTokenImport event
+     * @return array
+     */
+    protected function dispatchBeforeTokenImport($iSurveyId, $aToken, $sUploadCharset, $bFilterDuplicateToken, $bFilterBlankEmail, $bAllowInvalidEmail, $aFilterDuplicateFields, $sSeparator, $bShowWarningToken, $iRecordCount)
```

Comment:
Missing PHPDocs.

Reply:
Updated it

Reply:
I really dislike a bunch of params like this …

I really prefer something like 
````
public function dispatchBeforeTokenImport($iSurveyId, $aToken, $aOptions)
{
    $aOptions =array_merge(
        array(
            'csvcharset' => 'UTF-8',
            'filterduplicatetoken' => false,
            'filterblankemail' => true,
            'allowinvalidemail' => false,
            'filterduplicatefields' => array(),
            'separator' => ',',
            'showwarningtoken' => true,
        ),
        $aOptions
    );
    // Do the job
}

````
Really easiest to understand in my point of view.

Reply:
I agree. it is more readable. Let me re-write it. Honestly didn't want to create a block whre I need to setup that array just for calling this method.

Reply:
Thanks you :) And better it was send in array in plugin :)